components/src/core/components/CardTable/TableHeaderCell.vue

```diff
@@ -1,14 +1,59 @@
 <template>
   <div class="oxd-table-header-cell" role="columnheader">
     <slot></slot>
+    <oxd-icon-button
+      v-if="sort"
+      :withContainer="false"
+      :name="sortIcon"
+      class="oxd-table-header-cell-sorticon"
+      @click="$emit('sort', sort)"
+    />
   </div>
 </template>
 
 <script lang="ts">
-import {defineComponent} from 'vue';
+import {defineComponent, PropType} from 'vue';
+import {Sort} from './types';
+import IconButton from '@orangehrm/oxd/core/components/Button/Icon.vue';
 
 export default defineComponent({
   name: 'oxd-card-th',
+
+  props: {
+    sort: {
+      type: Object as PropType<Sort>,
+      required: false,
+    },
+  },
+
+  emits: ['sort'],
+
+  components: {
+    'oxd-icon-button': IconButton,
+  },
+
+  computed: {
+    sortIcon(): string {
+      let icon = '';
+      if (this.sort) {
+        switch (this.sort.state) {
+          case 'asc':
```

Comment:
Better to define `asc`, `desc`, `none` somewhere as consts. within types.ts file may be